xyz-hub-service/src/main/java/com/here/xyz/hub/task/SpaceTaskHandler.java

```diff
@@ -705,21 +706,43 @@ private static Future<Void> updateReadOnlyHeadVersion(Marker marker, Space space
     return p.future();
   }
 
-    public static void cleanDependentResources(ConditionalOperation task, Callback<ConditionalOperation> callback) {
-      if (task.isDelete()) {
-        String spaceId = task.responseSpaces.get(0).getId();
-        Service.tagConfigClient.deleteTagsForSpace(task.getMarker(), spaceId)
-            .onSuccess(a-> callback.call(task))
-            .onFailure(a->{
-              logger.error(task.getMarker(), "Failed to delete tags for space {}", spaceId, a);
-              callback.call(task);
-            });
-      }
-      else {
-        callback.call(task);
-      }
+  public static void cleanDependentResources(ConditionalOperation task, Callback<ConditionalOperation> callback) {
+    if (!task.isDelete()) {
+      callback.call(task);
+      return;
     }
 
+    final String spaceId = task.responseSpaces.get(0).getId();
+
+    final Future<List<Tag>> tagsFuture = Service.tagConfigClient.deleteTagsForSpace(task.getMarker(), spaceId)
+        .onFailure(e->logger.error(task.getMarker(), "Failed to delete tags for space {}", spaceId, e));
+
+    final Future<Void> deactivateFuture = getAllDependentSpaces(task.getMarker(), spaceId)
```

Comment:
Could be renamed to `#getAllExtendingSpaces(...)` see above.

Reply:
will be changed to getAllSuperSpaces